apps/test/unit/lib/kits/maker/boards/microBit/MicroBitBoardTest.js

```diff
@@ -27,200 +27,216 @@ function stubOpenSerialPort(board) {
-    board = new MicroBitBoard();
-    board.boardClient_ = new MicrobitStubBoard();
-    boardSetupAndStub(board);
-  });
-
-  afterEach(() => {
-    board = undefined;
-    sinon.restore();
-  });
-
-  describe('Maker Board Interface', () => {
-    itImplementsTheMakerBoardInterface(MicroBitBoard, board => {
+export function itMakesMicroBitComponentsAvailable(
+  Board,
+  boardSpecificSetup = null,
+  boardSpecificTeardown = null
+) {
+  /**
+   * After installing on the interpreter, test that the components and
+   * component constructors are available from the interpreter
+   */
+  describe('Micro Bit components accessible from interpreter', () => {
+    let jsInterpreter;
+    let board;
+
+    beforeEach(() => {
+      board = new Board();
       boardSetupAndStub(board);
-      sinon.stub(board.boardClient_, 'analogRead').callsArgWith(1, 0);
-      sinon.stub(board.boardClient_, 'digitalRead').callsArgWith(1, 0);
+
+      jsInterpreter = {
+        globalProperties: {},
+        createGlobalProperty: function (key, value) {
+          jsInterpreter.globalProperties[key] = value;
+        },
+        addCustomMarshalObject: sinon.spy()
+      };
+      // Opportunity to stub anything needed to test a board
+      if (boardSpecificSetup) {
+        boardSpecificSetup(board);
+      }
+
+      return board.connect();
     });
 
-    /**
-     * After installing on the interpreter, test that the components and
-     * component constructors are available from the interpreter
-     */
-    describe('Micro Bit components accessible from interpreter', () => {
-      let jsInterpreter;
-      let board;
+    afterEach(() => {
+      if (boardSpecificTeardown) {
+        boardSpecificTeardown(board);
+      }
+    });
 
+    describe('adds component constructors', () => {
       beforeEach(() => {
-        board = new MicroBitBoard();
-        boardSetupAndStub(board);
-
-        jsInterpreter = {
-          globalProperties: {},
-          createGlobalProperty: function (key, value) {
-            jsInterpreter.globalProperties[key] = value;
-          },
-          addCustomMarshalObject: sinon.spy()
-        };
-
-        return board.connect();
+        board.installOnInterpreter(jsInterpreter);
       });
 
-      describe('adds component constructors', () => {
-        beforeEach(() => {
-          board.installOnInterpreter(jsInterpreter);
-        });
+      it(`correct number of them`, () => {
+        expect(jsInterpreter.addCustomMarshalObject).to.have.callCount(
+          MB_COMPONENTS.length
+        );
+      });
 
-        it(`correct number of them`, () => {
-          expect(jsInterpreter.addCustomMarshalObject).to.have.callCount(
-            MB_COMPONENTS.length
+      MB_COMPONENTS.forEach(constructor => {
+        it(constructor, () => {
+          expect(jsInterpreter.globalProperties).to.have.ownProperty(
+            constructor
+          );
+          expect(jsInterpreter.globalProperties[constructor]).to.be.a(
```

Comment:
Big picture question: What risks would we see if we removed these tests?

Reasoning for asking: The main benefit I usually think of when it comes to unit tests is ensuring that branching conditions and business logic have protections from regressions. In this case, these tests seem to be checking whether each function exists, rather than what each function does. That makes me think this may be a section of code that doesn't need testing. Thoughts?

Reply:
👍